tap_helpshift/client.py

```diff
@@ -130,6 +131,11 @@ def __init__(self, session, config):
         subdomain = config['subdomain']
         self.base_url = 'https://api.helpshift.com/v1/{}'.format(subdomain)
         self.analytics_base_url = 'https://analytics.helpshift.com/v1/{}'.format(subdomain)
+        config_timeout = config.get('timeout')
+        if DEFAULT_TIMEOUT is not None:
+            self.timeout = max(config_timeout or 0, DEFAULT_TIMEOUT)
+        else:
+            self.timeout = config_timeout
```

Comment:
nit: since we're reading directly from the config maybe we constrain this to be a positive value? 

`config_timeout if (config_timeout and config_timeout >= 0) else None`

Reply:
So weird. requests would throw errors if it's a different type or negative, aiohttp throws an error if it's a different type. Will adjust this and raise an error if it's negative.

On a side note, it looks like aiohttp [defaults the timeout to 300s (5mins)](https://docs.aiohttp.org/en/stable/client_reference.html#aiohttp.ClientSession) which is different compared to requests which basically has an indefinite timeout.

So the asyncio.TimeoutError that was raised before we introduced `timeout` was because it times out after 300s (possibly because the request is big? don't expect it to be that large)